src/routine.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+func printFrom(from string) {
+	for i := 0; i < 3; i++ {
+		fmt.Println(from, ":", i)
+	}
+}
+
+func main() {
+	printFrom("direct")
+	go printFrom("goroutine")
+	go func(msg string) {
+		fmt.Println(msg)
+	}("going")
```

Comment:
Please write beautiful comments about why we need to pass the context to go routine. It is an important topic
also an example where a go routine is called in a for loop will be very helpful something like
```

for ...{

    go someFunc(){

}(context)

}

```

also show how one can use multiple procs using **GOMAXPROCS**